### PR TITLE
Test/mkl dnn act

### PR DESCRIPTION
**Pull request by azai91**
## Description ##
(Brief description on what this PR is about)

### Essentials ###
Please feel free to remove inapplicable items for your PR.
- [ ] The PR title starts with [MXNET-$JIRA_ID], where $JIRA_ID refers to the relevant [JIRA issue](https://issues.apache.org/jira/projects/MXNET/issues) created (except PRs with tiny changes)
- [ ] Changes are complete (i.e. I finished coding on this PR)
- [ ] All changes have test coverage:
- Unit tests are added for small changes to verify correctness (e.g. adding a new operator)
- Nightly tests are added for complicated/long-running ones (e.g. changing distributed kvstore)
- Build tests will be added for build configuration changes (e.g. adding a new build option with NCCL)
- [ ] Code is well-documented: 
- For user-facing API changes, API doc string has been updated. 
- For new C++ functions in header files, their functionalities and arguments are documented. 
- For new examples, README.md is added to explain the what the example does, the source of the dataset, expected performance on test set and reference to the original paper if applicable
- Check the API doc at http://mxnet-ci-doc.s3-accelerate.dualstack.amazonaws.com/PR-$PR_ID/$BUILD_ID/index.html
- [ ] To the my best knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change

### Changes ###
- [ ] Feature1, tests, (and when applicable, API doc)
- [ ] Feature2, tests, (and when applicable, API doc)

## Comments ##
- If this change is a backward incompatible change, why must this change be made.
- Interesting edge cases to note here

### CODE REVIEW

File: src/operator/quadratics_op-inl.h [hunk 1]
```diff
@@ -0,0 +1,31 @@
+//
```
Comment (anirudh2290) [inline src/operator/quadratics_op-inl.h]: Is this intentional?

**Comment by pengzhao-intel** [comment]
@juliusshufan is mkl act test on your test plan? you can work w/ @azai91 for this test.

File: tests/cpp/operator/mkldnn.cc [hunk 2]
```diff
@@ -353,6 +364,16 @@ OpAttrs GetCopyOp() {
   return attrs;
 }
 
+OpAttrs GetReluOp() {
+  OpAttrs attrs;
+  attrs.attrs.op = Op::Get("Activation");
+  attrs.attrs.dict.insert({"act_type", "relu"});
+  attrs.attrs.op->attr_parser(&attrs.attrs);
+  attrs.dispatches.resize(1);
+  attrs.dispatches[0] = DispatchMode::kFCompute;
```
Comment (zheng-da) [inline tests/cpp/operator/mkldnn.cc]: you need kFComputeEx here. MKLDNN uses kFComputeEx.

File: tests/cpp/operator/mkldnn.cc [hunk 3]
```diff
@@ -89,29 +91,42 @@ TEST(MKLDNN_UTIL_FUNC, MemFormat) {
...
 
 // Init arrays with the default layout.
-static void InitArray(NDArray *arr) {
+static void InitDefaultArray(NDArray *arr, bool is_rand = false) {
   const TBlob &blob = arr->data();
   mshadow::default_real_t *data = blob.dptr<mshadow::default_real_t>();
   size_t size = blob.Size();
-  for (size_t i = 0; i < size; i++)
-    data[i] = i;
+    if (is_rand) {
```
Comment (anirudh2290) [inline tests/cpp/operator/mkldnn.cc]: is_rand inside the for loop ?

File: tests/cpp/operator/mkldnn.cc [hunk 4]
```diff
@@ -89,29 +91,42 @@ TEST(MKLDNN_UTIL_FUNC, MemFormat) {
...
-// Init arrays with the specified layout.
-static void InitMKLDNNArray(NDArray *arr, const mkldnn::memory::primitive_desc &pd,
-                            bool is_rand = false) {
+// Init arrays with negative and positive values
+static void InitNegPosArray(NDArray *arr, bool is_rand = false) {
   const TBlob &blob = arr->data();
   mshadow::default_real_t *data = blob.dptr<mshadow::default_real_t>();
-  size_t size = blob.Size();
+  int size = blob.Size();
   if (is_rand) {
```
Comment (anirudh2290) [inline tests/cpp/operator/mkldnn.cc]: is_rand should be inside the for loop

**Comment by anirudh2290** [comment]
can you fix lint

File: src/operator/nn/mkldnn/mkldnn_sum.cc [hunk 5]
```diff
@@ -62,12 +62,9 @@ void MKLDNNSumForward(const nnvm::NodeAttrs& attrs, const OpContext &ctx,
...
   for (size_t i = 0; i < inputs.size(); i++) {
     const mkldnn::memory *in_mem;
-    if (inputs[i].IsMKLDNNData() && inputs[i].IsView()) {
+    if (inputs[i].IsMKLDNNData() && inputs[i].IsView())
       in_bufs[i] = inputs[i].Reorder2Default();
-      in_mem = in_bufs[i].GetMKLDNNData();
-    } else {
-      in_mem = inputs[i].GetMKLDNNData();
-    }
+    in_mem = inputs[i].GetMKLDNNData();
```
[the remark this answers is not in the document]
Comment (azai91) [reply src/operator/nn/mkldnn/mkldnn_sum.cc]: woops realized it was not the same. reverted.

File: tests/cpp/operator/mkldnn.cc [hunk 6]
```diff
@@ -598,51 +680,56 @@ void TestBinaryOp(const OpAttrs &attrs, VerifyFunc verify_fn) {
...
-      verify_fn(orig_inputs, arr);
+      verify_fn(orig_inputs, arr.arr);
     }
   }
 }
 
 TEST(IMPERATIVE, UnaryOp) {
   OpAttrs attrs = GetCopyOp();
-  TestUnaryOp(attrs, VerifyCopyResult);
+    TestUnaryOp(attrs, InitDefaultArray, VerifyCopyResult);
```
Comment (zheng-da) [inline tests/cpp/operator/mkldnn.cc]: please fix the indent here.

**Pull request merged**